test/functional/sqlsrv/sqlsrv_escape_braces.phpt

```diff
@@ -0,0 +1,70 @@
+--TEST--
+Test that right braces are escaped correctly and that error messages are correct when they're not
+--SKIPIF--
+<?php require('skipif.inc'); ?>
+--FILE--
+<?php
+$server = 'fakeserver';
+$uid = 'sa';
+$password = 'fakepassword';
+
+// If the braces are fine, then we expect the connection to fail with a login timeout error
+$braceError = "An unescaped right brace (}) was found";
+$connError = (strtoupper(substr(php_uname('s'), 0, 3)) === 'WIN') ? "Could not open a connection to SQL Server" : "Login timeout expired";
+
+// Every combination of one, two, three, or more right braces I can think of
+$testStrings = array(array("}", $braceError),
```

Comment:
As shown in #905 please add test cases with strings that consist of punctuation and/or space characters?

Reply:
That doesn't really have anything to do with escaping braces though - the code changes only affect braces. Should there be a separate test for nonalphanumeric characters in strings?

Reply:
Ok you can leave that. I think we have a backlog item created already